queries.sql

```diff
@@ -1,3 +1,97 @@
 /*Queries that provide answers to the questions from all projects.*/
 
-SELECT * from animals WHERE name = 'Luna';
+
+-- Query for Find all animals whose name ends in "mon".
+SELECT * FROM animals WHERE name LIKE '%mon';
+
+-- Query for List the name of all animals born between 2016 and 2019.
+SELECT name FROM animals
+WHERE date_of_birth BETWEEN '2016-1-1'AND '2019-12-31';
+
+-- Query for List the name of all animals that are neutered and have less than 3 escape attempts.
+SELECT name FROM animals
+WHERE neutered = true  AND escape_attempts < 3;
+
+-- Query for List the date of birth of all animals named either "Agumon" or "Pikachu".
+SELECT date_of_birth FROM animals
+WHERE name = 'Agumon' OR name ='Pikachu' ;
+
+-- Query for List name and escape attempts of animals that weigh more than 10.5kg
+SELECT name,escape_attempts FROM animals
+WHERE weight_kg > 10.5 ;
+
+
+-- Query for Find all animals that are neutered.
+SELECT * FROM animals
+WHERE neutered = true ;
+
+-- Query for Find all animals not named Gabumon.
+SELECT * FROM animals
+WHERE name <> 'Gabumon' ;
+
+-- Query for Find all animals with a weight between 10.4kg and 17.3kg
+SELECT * FROM animals
+WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+--Query for transaction to set species ->digimon to animals ending in mon and species->pokemon to animals with no species
+BEGIN;
+UPDATE animals
+SET species ='digimon'
+WHERE name LIKE '%mon';
+
+UPDATE animals
+SET species ='pokemon'
+WHERE species IS NULL;
+COMMIT;
+
+--Query for transaction to delete all data and rollback
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+--Query for transacton to delete animals with dob> 2022 making savepoint and update weight of animal with negative weight
+BEGIN;
+DELETE FROM animals
+WHERE date_of_birth >'2022-01-01';
+SAVEPOINT born_before;
+UPDATE animals
+SET weight_kg = weight_kg * (-1);
+ROLLBACK TO born_before;
+UPDATE animals
+SET weight_kg = weight_kg * (-1)
+WHERE weight_kg < -1;
```

Comment:
- [ ] Kindly update this query inside the transaction to match the [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Update%20all%20animals%27%20weights%20that%20are%20negative%20to%20be%20their%20weight%20multiplied%20by%20%2D1.). You need to update all animals' weights that are negative to be their weight multiplied by -1. There is a problem with the where clause. Kindly update the where clause to check for all negative values as now will not check for the negative values from 0 to -1.

### Hint: 
Update the where clause to `WHERE weight_kg < 0` instead of `WHERE weight_kg < -1`